### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by Bifabig**
In this PR I have made a test for posts and users URL requests
- checks if response status was correct.
- checks if a correct template was rendered and the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/post_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    it 'correct response status for /show' do
+      get '/users/1/posts/1'
+      expect(response).to be_successful
+    end
+
+    it 'response body includes correct placeholder text for show' do
+      get '/users/1/posts/1'
+      expect(response.body).to include('Here is a list of posts for a given user!')
+    end
+  end
```
Comment (julie-ify) [inline spec/requests/post_request_spec.rb]: - [x] Kindly separate the index and show tests using the describe block.

  ```
  describe 'GET /index' do
    ...
  end
  
  describe 'GET /show' do
    ...
  end
  ```

File: spec/requests/user_request_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    it 'correct response status for /show' do
+      get '/users/1'
+      expect(response).to be_successful
+    end
+
+    it 'response body includes correct placeholder text for /show' do
+      get '/users/1'
+      expect(response.body).to include('Here are details about the User!')
+    end
+  end
```
Comment (julie-ify) [inline spec/requests/user_request_spec.rb]: - [x] Kindly separate the index and show tests using the describe block.

  ```
  describe 'GET /index' do
    ...
  end
  
  describe 'GET /show' do
    ...
  end
  ```

File: spec/requests/user_request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  describe 'GET ' do
+    it 'correct response status for /index' do
+      get '/users'
+      expect(response).to be_successful
+    end
+
+    it 'response body includes correct placeholder text for /index' do
+      get '/users'
+      expect(response.body).to include('Users are coming soon!')
+    end
```
Comment (julie-ify) [inline spec/requests/user_request_spec.rb]: - [x] For the users' index test, kindly include a test that checks that the index template is rendered.

  ```
  expect(response).to render_template(:index)
  ```

File: spec/requests/post_request_spec.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    it 'returns http success for /index' do
+      get '/users/1/posts'
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'response body includes correct placeholder text for /index' do
+      get '/users/1/posts'
+      expect(response.body).to include('Posts are coming soon!')
+    end
+
```
Comment (julie-ify) [inline spec/requests/post_request_spec.rb]: - [x] For the posts' index test, kindly include a test that checks that the index template is rendered.

  ```
  expect(response).to render_template(:index)
  ```

File: spec/requests/post_request_spec.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    it 'correct response status for /show' do
+      get '/users/1/posts/1'
+      expect(response).to be_successful
+    end
+
+    it 'response body includes correct placeholder text for show' do
+      get '/users/1/posts/1'
+      expect(response.body).to include('Here is a list of posts for a given user!')
+    end
```
Comment (julie-ify) [inline spec/requests/post_request_spec.rb]: - [x] For the posts' show test, kindly include a test that checks that the show template is rendered.

  ```
  expect(response).to render_template(:show)
  ```

File: spec/requests/user_request_spec.rb (hunk 2 above)
Comment (julie-ify) [inline spec/requests/user_request_spec.rb]: - [x] For the users' show test, kindly include a test that checks that the show template is rendered.

  ```
  expect(response).to render_template(:show)
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Bifabig,

Good job so far!

## Highlights!
- Successful test cases
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**